lib/library.rb

```diff
@@ -0,0 +1,31 @@
+class Library
+  attr_reader :name, :books, :authors
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+  end
+
+  def add_author(author)
+    @authors << author
+    author.books.each do |book|
+      @books << book
+    end
+  end
+
+ # can be cleaned up with helper methods if there is time...
```

Comment:
Totally - the logic here is great, but it could be broken up into helper methods.  Also, what might it look like if an author could give you a list of their books sorted by publication year?